src/System.Net.Http/src/System/Net/Http/Headers/ContentDispositionHeaderValue.cs

```diff
@@ -564,18 +565,27 @@ private string Encode5987(string input)
         private bool TryDecode5987(string input, out string output)
         {
             output = null;
-            string[] parts = input.Split('\'');
-            if (parts.Length != 3)
+            
+            int quoteIndex = input.IndexOf('\'');
+            if (quoteIndex == -1)
             {
                 return false;
             }
+            
+            int lastQuoteIndex = input.LastIndexOf('\'');
+            if (quoteIndex == lastQuoteIndex || input.IndexOf('\'', quoteIndex + 1) != lastQuoteIndex)
+            {
+                return false;
```

Comment:
Are there any tests that hit this?  I don't see any in either the functional or unit tests.